components/payments/cmd/connectors/internal/connectors/atlar/task_initiate_payment.go

```diff
@@ -0,0 +1,145 @@
+package atlar
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/formancehq/payments/cmd/connectors/internal/connectors/currency"
+	"github.com/formancehq/payments/cmd/connectors/internal/ingestion"
+	"github.com/formancehq/payments/cmd/connectors/internal/metrics"
+	"github.com/formancehq/payments/cmd/connectors/internal/storage"
+	"github.com/formancehq/payments/cmd/connectors/internal/task"
+	"github.com/formancehq/payments/internal/models"
+	"github.com/formancehq/stack/libs/go-libs/logging"
+	atlar_client "github.com/get-momo/atlar-v1-go-client/client"
+	"github.com/get-momo/atlar-v1-go-client/client/credit_transfers"
+	atlar_models "github.com/get-momo/atlar-v1-go-client/models"
+)
+
+func InitiatePaymentTask(config Config, client *atlar_client.Rest, transferID string) task.Task {
+	return func(
+		ctx context.Context,
+		logger logging.Logger,
+		connectorID models.ConnectorID,
+		resolver task.StateResolver,
+		scheduler task.Scheduler,
+		storageReader storage.Reader,
+		ingester ingestion.Ingester,
+		metricsRegistry metrics.MetricsRegistry,
+	) error {
+		transferInitiationID := models.MustTransferInitiationIDFromString(transferID)
+		transfer, err := getTransfer(ctx, storageReader, transferInitiationID, true)
+		if err != nil {
+			return err
+		}
+
+		err = validateTransferInitiation(transfer)
+		if err != nil {
+			return err
+		}
+
+		currency, precision, err := currency.GetCurrencyAndPrecisionFromAsset(supportedCurrenciesWithDecimal, transfer.Asset)
+		if err != nil {
+			return err
+		}
+
+		paymentSchemeType := "SCT" // SEPA Credit Transfer
+		remittanceInformationType := "UNSTRUCTURED"
+		remittanceInformationValue := transfer.Description
+		amount := atlar_models.AmountInput{
+			Currency:    &currency,
+			Value:       transfer.Amount.Int64(),
+			StringValue: amountToString(*transfer.Amount, precision),
+		}
+		date := transfer.ScheduledAt
+		if date.IsZero() {
+			date = time.Now()
+		}
+		dateString := date.Format(time.DateOnly)
+		logger.WithContext(ctx).Infof("date is %s", date.Format(time.RFC3339Nano))
+
+		createPaymentRequest := atlar_models.CreatePaymentRequest{
+			SourceAccountID:              &transfer.SourceAccount.Reference,
+			DestinationExternalAccountID: &transfer.DestinationAccount.Reference,
+			Amount:                       &amount,
+			Date:                         &dateString,
+			ExternalID:                   transfer.ID.Reference, // TODO: don't know if that's correct
```

Comment:
If it corresponds to a unique request id for Atlar, then yes !

Reply:
I think you should concatenate the reference with the number of attempts (that you have in the transfer structure also). That way, if you want to retry a payment, it will not have the same external id

Reply:
done

Reply:
> I think you should concatenate the reference with the number of attempts (that you have in the transfer structure also). That way, if you want to retry a payment, it will not have the same external id

Sorry, GitHub did not show this comment when I posted my replies.

But as I already wrote in chat, I came to realize it might make more sense to post the payments to Atlar asynchronously with retries, since we need to check for success asynchronously anyways. It only makes sense to move money in the ledger if the payment has been successful or if it has failed.